Apsitvarkom.UnitTests/Api/Controllers/PollutedLocationControllerTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Apsitvarkom.Api.Controllers;
+using Apsitvarkom.DataAccess;
+using Apsitvarkom.Models.DTO;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+
+namespace Apsitvarkom.UnitTests.Api.Controllers;
+
+public class PollutedLocationControllerTests
+{
+    private Mock<IPollutedLocationDTORepository> m_repository;
+    private PollutedLocationController m_controller;
+
+    [SetUp]
+    public void SetUp()
+    {
+        m_repository = new Mock<IPollutedLocationDTORepository>();
+        m_controller = new PollutedLocationController(m_repository.Object);
+    }
+
+    [Test]
+    public void Constructor_HappyPath_IsSuccess() => Assert.That(new PollutedLocationController(m_repository.Object), Is.Not.Null);
+
+    [Test]
+    public async Task GetAll_RepositoryReturnsPollutedLocationDTOs_OKActionResultReturned()
+    {
+        var instances = GetPollutedLocationDTOs();
+        m_repository.Setup(self => self.GetAllAsync()).ReturnsAsync(instances);
+
+        var actionResult = await m_controller.GetAll();
+
+        var result = actionResult.Result as OkObjectResult;
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result!.StatusCode, Is.EqualTo(200));
+        Assert.That(result.Value, Is.EqualTo(instances));
+    }
+
+    [Test]
+    public async Task GetById_RepositoryReturnsPollutedLocationDTO_OKActionResultReturned()
+    {
+        var instanceId = Guid.NewGuid().ToString();
+        var instance = new PollutedLocationDTO
```

Comment:
Any reasons why create a new object here and not for example `GetPollutedLocationDTOs[0]`?

Reply:
or `.First()`